### PR TITLE
Config option to set sync interval

### PR DESCRIPTION
**Pull request by tokun-mobica**
PR related to [Suggestion: Config option to set sync interval #2792 ](https://github.com/nextcloud/desktop/issues/2792)
Requested sync interval can be set with "localSyncInterval" variable in [Nextcloud] section of client configuration file.
Passed value denotes miliseconds, default is 0 if no proper config record is present.

### CODE REVIEW

File: src/gui/folderwatcher.cpp [hunk 1]
```diff
@@ -149,6 +152,19 @@ void FolderWatcher::changeDetected(const QString &path)
     changeDetected(paths);
 }
 
+void FolderWatcher::notifyChangedPaths()
+{
+    _syncInterval.stop();
+    _lastPaths.clear();
+    if (!_changedPaths.isEmpty()) {
```
Comment (mgallien) [inline src/gui/folderwatcher.cpp:159]: when can we end up in this method with an empty `_changedPaths` ?

**Review by mgallien — COMMENTED** [review]
thanks

File: src/gui/folderwatcher.cpp (hunk 1 above)
Comment (tokun-mobica) [reply src/gui/folderwatcher.cpp]: Good point, that shouldn't happen.

**Comment by tokun-mobica** [comment]
@mgallien Thanks for the review!
I've updated code taking all remarks under account

**Review by mgallien — CHANGES_REQUESTED** [review]
going to need more time for a proper review as I do not want to impact most users with a change that could generate bugs for them
please see my small comment

**Comment by mgallien** [comment]
thanks @tokun-mobica 
I promise to not forget about this and sorry for the delays

File: src/libsync/configfile.h [hunk 2]
```diff
@@ -75,6 +75,8 @@ class OWNCLOUDSYNC_EXPORT ConfigFile
 
     /* Force sync interval, in milliseconds */
     std::chrono::milliseconds forceSyncInterval(const QString &connection = QString()) const;
+    /* Local files sync interval, in milliseconds */
+    std::chrono::milliseconds localSyncInterval(const QString &connection = QString()) const;
```
Comment (allexzander) [inline src/libsync/configfile.h:79]: @tokun-mobica @tokun-mobica `const QString &connection = {}`

File: src/libsync/configfile.cpp [hunk 3]
```diff
@@ -542,6 +544,23 @@ chrono::milliseconds ConfigFile::forceSyncInterval(const QString &connection) co
     return interval;
 }
 
+chrono::milliseconds ConfigFile::localSyncInterval(const QString &connection) const
+{
+    QString con(connection);
```
Comment (allexzander) [inline src/libsync/configfile.cpp:549]: @tokun-mobica `auto con = connection`

**Comment by tokun-mobica** [comment]
@allexzander Thanks for the review! Code has been updated

File: src/gui/folderwatcher.cpp [hunk 4]
```diff
@@ -43,6 +45,8 @@ FolderWatcher::FolderWatcher(Folder *folder)
     : QObject(folder)
     , _folder(folder)
 {
+    _syncIntervalLenght = ConfigFile().localSyncInterval();
```
Comment (claucambra) [inline src/gui/folderwatcher.cpp:48]: Please fix the typo :)

**Comment by claucambra** [comment]
Closes #2792

**Review by claucambra — APPROVED** [review]
Really tiny comment, besides that it looks good to me

File: src/gui/folderwatcher.cpp (hunk 4 above)
Comment (tokun-mobica) [reply src/gui/folderwatcher.cpp]: Thanks, Hawkeye ;)

**Comment by Rello** [comment]
Hello,

Thank you for your contribution to the Desktop Client with this pull request.
We are closing this request as it is outdated, no longer relevant (e.g., due to Qt 6), or does not align with the current roadmap.

In this special case we recommend not to rely on polling but using the notify_push implementation for much better performance.
We truly value your support and hope to see more contributions from you in the future!


Best regards,
Nextcloud Desktop Team

**Pull request closed without merging**